src/app/workspace/organization/detail/payments-panel/payments-list/payments-list.component.ts

```diff
@@ -17,18 +17,85 @@
+import {isNullOrUndefined} from "util";
+import {filter} from "rxjs/operators";
+import {PaymentModel} from "../../../../../core/store/organizations/payment/payment.model";
+import {selectPaymentsByOrganizationIdSorted} from "../../../../../core/store/organizations/payment/payments.state";
+import {PaymentsAction, PaymentsActionType} from "../../../../../core/store/organizations/payment/payments.action";
+import {ServiceLimitsAction} from "../../../../../core/store/organizations/service-limits/service-limits.action";
 
 @Component({
   selector: 'payments-list',
   templateUrl: './payments-list.component.html',
   styleUrls: ['./payments-list.component.scss']
 })
-export class PaymentsListComponent implements OnInit {
+export class PaymentsListComponent implements OnInit, OnDestroy {
 
-  constructor() { }
+  private organization: OrganizationModel;
+  private organizationSubscription: Subscription;
+
+  public payments: PaymentModel[];
+  private paymentsSubscription: Subscription;
+
+  private getPaymentSubscription: Subscription;
+
+  constructor(private i18n: I18n,
+              private router: Router,
+              private store: Store<AppState>,
+              private actionsSubject: ActionsSubject) { }
 
   public ngOnInit() {
+    this.subscribeToStore();
+    this.subscribeToActions();
+    this.requestData();
+  }
+
+  public subscribeToStore() {
+    this.organizationSubscription = this.store.select(selectOrganizationByWorkspace)
+      .pipe(filter(organization => !isNullOrUndefined(organization)))
+      .subscribe(organization => this.organization = organization);
+
+    this.paymentsSubscription = this.store.select(selectPaymentsByOrganizationIdSorted(this.organization.id))
+      .pipe(filter(payments => !isNullOrUndefined(payments) && payments.length > 0))
+      .subscribe(payments => this.payments = payments);
+  }
+
+  public requestData() {
+    this.store.dispatch(new PaymentsAction.GetPayments({ organizationId: this.organization.id }));
   }
 
+  public ngOnDestroy(): void {
+    if (this.organizationSubscription) {
+      this.organizationSubscription.unsubscribe();
+    }
+
+    if (this.paymentsSubscription) {
+      this.paymentsSubscription.unsubscribe();
+    }
+
+    if (this.getPaymentSubscription) {
+      this.getPaymentSubscription.unsubscribe();
+    }
+  }
+
+  public refreshPayment(paymentId: string) {
+    this.store.dispatch(new PaymentsAction.GetPayment({ organizationId: this.organization.id, paymentId }));
+  }
+
+  private subscribeToActions() {
+    this.getPaymentSubscription = this.actionsSubject.subscribe(data => {
+      if (data.type === PaymentsActionType.GET_PAYMENT_SUCCESS) {
+        // we have a new payment state obtained through manual button, refresh service limits as well
+        this.store.dispatch(new ServiceLimitsAction.GetServiceLimits({ organizationId: this.organization.id }));
```

Comment:
What if user navigates to some other page before `GET_PAYMENT_SUCCESS` action is dispatched? The service limits will not be updated in such case since this subscription will be destroyed together with this component.

Better approach is to have an optional `nextAction?: Action` parameter in the payload of `GetPayment` action and dispatch this `nextAction` (if defined) together with `GetPaymentSuccess` action in the effects class. Then in this component you can simply pass `new ServiceLimitsAction.GetServiceLimits({ organizationId: this.organization.id })` as `nextAction` parameter.

Reply:
I really cannot do that. First, I would need this nextAction in CreatePayment, not GetPayment. However, I do not have access to action object in the phase of mapping the output of calling http service.  Any workaround?

Reply:
Why `CreatePaymentAction`? In the code above you are clearly reacting to `GetPaymentAction` and not `CreatePaymentAction`.

When it comes to the `action` object not being available after the HTTP service call, just look at other effect classes how we pass this object further down the stream.